lib/ui/painting.dart

```diff
@@ -1595,6 +1640,60 @@ class Image extends NativeFieldWrapperClass2 {
   ///
   /// Returns a future that completes with the binary image data or an error
   /// if encoding fails.
+  Future<ByteData?> toByteData({ImageByteFormat format = ImageByteFormat.rawRgba}) {
+    assert(!_disposed && !_image._disposed);
+    return _image.toByteData(format: format);
+  }
+
+  bool _disposed = false;
```

Comment:
Should be named _debugDisposed since its only used within asserts and will only be valid in debug mode.

Reply:
Also, move this closer to the dispose method?

Reply:
Moved it, but also moved setting it out of asserts, since I think the runtime failure mode without it will be pretty confusing.